tutorials/polkadot-sdk/parachains/build-custom-pallet/build-pallet.md

```diff
@@ -0,0 +1,290 @@
+A pallet encapsulates a specific set of blockchain functionalities, such as managing token balances, implementing governance mechanisms, or creating custom state transitions.
+
+In this tutorial, you'll learn how to create a custom pallet from scratch. You will develop a simple counter pallet with the following features:
+
+- Users can increment and decrement a counter
+- Only a [root origin](https://paritytech.github.io/polkadot-sdk/master/frame_system/pallet/type.Origin.html#variant.Root){target=\_blank} can set an arbitrary counter value
+
+You'll use the [Polkadot SDK Solochain Template](https://github.com/paritytech/polkadot-sdk-solochain-template){target=\_blank}, a pre-configured blockchain template that provides a functional development environment.
+
+## Prerequisites
+
+To set up your development environment for the Polkadot SDK, you'll need:
+
+- Rust installation - the node template is written in [Rust](https://www.rust-lang.org/){target=\_blank}. Install it by following the [Installation](/develop/parachains/get-started/install-polkadot-sdk){target=\_blank} guide for step-by-step instructions on setting up your development environment
+
+## Set Up a Chain Template
+
+The [Polkadot SDK Solochain Template](https://github.com/paritytech/polkadot-sdk-solochain-template){target=\_blank} provides a ready-to-use development environment for building using the [Polkadot SDK](https://github.com/paritytech/polkadot-sdk){target=\_blank}. Follow these steps to compile the node:
+
+1. Clone the repository:
+    ```bash
+    git clone -b {{dependencies.polkadot_sdk_solochain_template.version}} {{dependencies.polkadot_sdk_solochain_template.repository_url}}
+    ```
+
+    !!!note
+        Ensure you're using the version `{{dependencies.polkadot_sdk_solochain_template.version}}` of the Polkadot SDK Solochain Template to be able to follow this tutorial step by step.
+
+2. Navigate to the project directory:
+    ```bash
+    cd polkadot-sdk-solochain-template
+    ```
+
+3. Compile the node template:
+    ```bash
+    cargo build --release
+    ```
+
+    !!!note
+        Depending on your machine's specifications, initial compilation may take several minutes. For optimized artifacts, use the `--release` flag.
+
+4. Verify successful compilation by checking that the output is similar to:
+    --8<-- 'code/tutorials/polkadot-sdk/parachains/build-custom-pallet/build-pallet/compilation-output.html'
+
+## Create a New Project
+
+In this tutorial, you'll build a custom pallet from scratch to demonstrate the complete workflow, rather than starting with the pre-built pallet-template. The first step is to create a new Rust package for your pallet:
```

Comment:
is pallet-template a thing? like a directory or something?

Reply:
Yes. When you download a node template like `solochain-template`, it comes with a predefined structure that includes a `pallets` directory. Within this directory, there's a `template` folder containing a scaffolded Rust project called `pallet-template`, which serves as a starting point for building custom pallets.

Reply:
```suggestion
In this tutorial, you'll build a custom pallet from scratch to demonstrate the complete workflow, rather than starting with the pre-built `pallet-template`. The first step is to create a new Rust package for your pallet:
```

Ah ok, then we should just put it in backticks